src/chapters/aspdotnet-model-classes/studio.rst

```diff
@@ -0,0 +1,110 @@
+#. Create ``Index.cshtml`` and ``Add.cshtml`` templates within ``Views/User/`` 
+#. Create a ``User`` class within ``Models``
+
+Creating the Model
+------------------
+
+Your ``User`` class should have a few properties: ``Username``, ``Email``, and ``Password``. 
+
+Rendering the Add User Form
+---------------------------
+
+#. In the ``UserController``, create an action method ``Add()`` to
+   render the form. This handler should correspond to the path
+   ``/user/add``, and for now, it can just return the ``Add.cshtml`` view.
+
+   .. admonition:: Tip
+
+      Don't forget to add ``/user/add`` to your path when you test your new features. 
+
+#. Within the ``Add.cshtml`` template, create a form that accepts inputs for
+   each of the ``User`` class properties. Include an additional password input field to verify 
+   the password input. The form should be set up to ``POST`` to ``/user``. 
+
+#. Be sure to set ``type="password"`` for the password and verify inputs,
+   to ensure the passwords are not visible when being typed into the form.
+   You can also set ``type="email"`` on the email input, which will enable
+   some basic client-side validation. We'll tackle validation in more detail 
+   in the next studio. 
+
+Processing Form Submission
+--------------------------
+
+#. Within the ``UserController``, create an action method with this signature:
+
+   .. sourcecode:: csharp
+
+      public IActionResult AddUser(User newUser, string verify) {
+         // add form submission handling code here
+      }
+
+   This will use model binding to create a new user object, ``newUser``, and
+   pass it into your action method. 
+
+   .. admonition :: Note
+   
+      You don’t need to store the ``User`` object anywhere for this studio.
+      We’re focusing on form handling and validation in this exercise. If you
+      want to keep track of users using the method we employed in the models
+      lesson video, check out the Bonus Missions below.
+
+#. Check that the ``verify`` parameter matches the
+   password within the ``newUser`` object. If it does, render the
+   ``Index.cshtml`` view template with a message that welcomes the user by 
+   username. If the passwords don’t match, render the form again.
+
+   a. To render the ``Index.cshtml`` view with the welcome message, you need to pass in the username as part of the route for ``Redirect()``.
+      You can do so using string concatenation: ``"/user?name=" + newUser.Username``. This will pass in the username as a query parameter.
+
+   b. Passing in the value to the ``Index.cshtml`` view also requires you to modify ``Index()``. You will need to add a parameter and use ``ViewBag``.
+
+Refining Form Submission
+------------------------
+
+#. Once registered, we want the user to access the form selecting their spa services. 
+
+   a. In ``Index.cshtml``, add an ``a`` element to take the user back to the path, ``/spa``, where the ``Spa/Index.cshtml`` template will be rendered.
```

Comment:
Just to add a few more details here - i'd clarify that the first index view mentioned is user/index. and probably not a bad idea to nudge them to use the asp-controller and asp-action directives, no?

the error processing instructions after this look good, btw. Just a suggestion though, can the solution code include smaller conditional blocks that display the error messages rather than repeating the whole form?